src/lib/directive.ts

```diff
@@ -26,26 +26,45 @@ export type DirectiveFactory = (...args: any[]) => object;
 export type DirectiveFn = (part: Part) => void;
 
 /**
- * Brands a function as a directive so that lit-html will call the function
- * during template rendering, rather than passing as a value.
+ * Brands a function as a directive factory function so that lit-html will call
+ * the function during template rendering, rather than passing as a value.
+ *
+ * A _directive_ is a function that takes a Part as an argument. It has the
+ * signature: `(part: Part) => void`.
+ *
+ * A directive _factory_ is a function that takes arguments for data and
+ * configuration and returns a directive. Users of directive usually refer to
+ * the directive factory as the directive. For example, "The repeat directive".
```

Comment:
Not a comment for the present, but this paragraph (and this naming scheme) makes for really confusing reading. We should discuss whether there's some way to make this less confusing. Maybe there isn't, dunno.